Visualization/generate_BackBLAST_heatmap.R

```diff
@@ -0,0 +1,618 @@
+plot_ggtree <- function(phylo_tree, bootstrap_label_data) {
+  
+  tree_plot <- ggtree::ggtree(phylo_tree, size = 1, colour = "black", ladderize = TRUE,
+                      branch.length = 0.1) +
+    
+    # Add the bootstrap labels from the external file
+    geom_text(data = bootstrap_label_data, aes(label = label), nudge_x = -0.035, nudge_y = 0.35, size = 3) +
+    
+    # Add the dotted lines from the tree tips
+    geom_tiplab(align = TRUE, linetype = "dotted", size = 0.1, offset = 0.1) +
+    
+    # Add scale bar
+    geom_treescale(x = 0, y = 8, linesize = 1, fontsize = 3, offset = 0.5) +
+    
+    # Manually tune the y-axis boundaries to match the heatmap: https://stackoverflow.com/a/18718196 (accessed Sept. 15, 2018)
+    # TODO - this might need to be a function of the number of entries
+    scale_y_discrete(expand = c(0,0.6)) +
+  
+    ## Manually set the righthand cutoff for the tree
+    xlim_tree(1)
+    # xlim etc.: https://guangchuangyu.github.io/2016/10/xlim_tree-set-x-axis-limits-for-only-tree-panel/ (accessed Sept 15, 2018)
+  
+  return(tree_plot)
+  
+}
+
+
+#' Master function to load and plot the phylogenetic tree
+#' 
+#' @param input_phylogenetic_tree_filepath Character (length 1 vector); the filepath to the phylogenetic tree
+#' @param root_name Character (length 1 vector); the exact name of the to-be root
+#' @param bootstrap_cutoff Numeric (length 1 vector); the minimum bootstrap cutoff, in percent
+#' @return list of two: 'phylo_tree' - ggtree unplotted object ; 'phylo_tree_fig' - ggtree figure
+#' @export
+load_and_plot_phylogenetic_tree <- function(input_phylogenetic_tree_filepath, root_name, bootstrap_cutoff) {
+  # Read tree
+  futile.logger::flog.info("Reading input phylogenetic tree")
+  phylo_tree <- ape::read.tree(input_phylogenetic_tree_filepath)
+  
+  # Optionally re-root tree
+  if ( is.na(root_name) == FALSE && root_name != "NA" ) {
+    futile.logger::flog.info(glue::glue("Re-rooting tree to '", root_name, "'"))
+    phylo_tree <- reroot_ggtree(phylo_tree, root_name)
+  }
+  
+  # Set cutoff for bootstraps externally, to be overlaid onto the tree figure later
+  # No cutoff is applied if bootstrap_cutoff is set to 'NA'
+  futile.logger::flog.debug("Generating bootstrap labels")
+  bootstrap_label_data <- generate_bootstrap_labels(phylo_tree, bootstrap_cutoff)
+  
+  # Generate tree plot
+  futile.logger::flog.info("Generating ggtree plot")
+  phylo_tree_fig <- plot_ggtree(phylo_tree, bootstrap_label_data)
+  
+  # Make list to return to user
+  tree_list <- list(phylo_tree, phylo_tree_fig)
+  names(tree_list) <- c("phylo_tree", "phylo_tree_fig")
+  
+  return(tree_list)
+  
+}
+
+
+#' Loads BLAST table and checks for expected column names (HARD-CODED in function)
+#' 
+#' @param input_blast_table_filepath Character (length 1 vector); the filepath to the BLAST table (comma-separated)
+#' @return tibble of BLAST data
+#' @export
+read_blast_tibble <- function(input_blast_table_filepath) {
```

Comment:
Change this to something like ```read_blast_results```. Naming it tibble means that we're stuck with that specific data type. For example lets say there's some far spangled new version of tibble made for gpus called gibble and we replace all the tibbles with gibbles. If the name is ```read_blast_tibble``` next developer is going to be confused because hes expecting a tibble not a gibble based off the name.

Reply:
Done -- thanks for the tip